gensim/downloader.py

```diff
@@ -0,0 +1,437 @@
+    else:
+        url_data = "https://github.com/RaRe-Technologies/gensim-data/releases/download/{f}/{f}.gz".format(f=name)
+        fname = "{f}.gz".format(f=name)
+        dst_path = os.path.join(tmp_dir, fname)
+        urllib.urlretrieve(url_data, dst_path, reporthook=_progress)
+        if _calculate_md5_checksum(dst_path) == _get_checksum(name):
+            sys.stdout.write("\n")
+            sys.stdout.flush()
+            logger.info("%s downloaded", name)
+        else:
+            shutil.rmtree(tmp_dir)
+            raise Exception("Checksum comparison failed, try again")
+        os.rename(tmp_dir, data_folder_dir)
+
+
+def _get_filename(name):
+    """Retrieve the filename of the dataset/model.
+
+    Parameters
+    ----------
+    name: str
+        Name of dataset/model.
+
+    Returns
+    -------
+    str:
+        Filename of the dataset/model.
+
+    """
+    information = info()
+    corpora = information['corpora']
+    models = information['models']
+    if name in corpora:
+        return information['corpora'][name]["file_name"]
+    elif name in models:
+        return information['models'][name]["file_name"]
+
+
+def load(name, return_path=False):
+    """Download (if needed) dataset/model and load it to memory (managed by return_path).
+
+    Parameters
+    ----------
+    name: str
+        Name of the model/dataset.
+    return_path: bool, optional
+        If True, return full path to file, otherwise, return loaded model / iterable dataset.
+
+    Returns
+    -------
+    Model
+        Requested model, if `name` is model and `return_path` == False.
+    Dataset (iterable)
+        Requested dataset, if `name` is dataset and `return_path` == False.
+    str
+        Path to file with dataset / model, only when `return_path` == True.
+
+    Raises
+    ------
+    Exception
+        Raised if `name` is incorrect.
+
+    Examples
+    --------
+    Model example:
+
+    >>> import gensim.downloader as api
+    >>>
+    >>> model = api.load("glove-twitter-25")  # load glove vectors
+    >>> model.most_similar("cat")  # show words that similar to word 'cat'
+
+    Dataset example:
+
+    >>> import gensim.downloader as api
+    >>>
+    >>> wiki = api.load("wiki-en")  # load extracted Wikipedia dump, around 6 Gb
+    >>> for article in wiki:  # iterate over all wiki script
+    >>>     ...
+
+    Download only example
+    >>> import gensim.downloader as api
+    >>>
+    >>> print(api.load("wiki-en", return_path=True))  # output: /home/user/gensim-data/wiki-en/wiki-en.gz
+
+    """
+    _create_base_dir()
+    file_name = _get_filename(name)
+    if file_name is None:
+        raise Exception(
+            "Incorrect model/corpus name. Choose the model/corpus from the list "
+            "\n {}".format(json.dumps(info(), indent=4))
+        )
+    folder_dir = os.path.join(base_dir, name)
+    path = os.path.join(folder_dir, file_name)
+    if not os.path.exists(folder_dir):
+        _download(name)
```

Comment:
Any way to "force" a download? (fix previously broken / partial / deleted downloads etc)

Reply:
This situation doesn't happen (because data moved from `/tmp/..` to `~/gensim-data` **after** downloading needed files + checking md5). For this reason, we don't add this key (if something happens - run `load` again.